recipe/meta.yaml

```diff
@@ -5,13 +5,47 @@ package:
 build:
   number: 0
 
-test:
-  commands:
-    # Verify the scripts are in-place.
-    {% for state in ["activate", "deactivate"] %}
-    - test -f "${PREFIX}/etc/conda/{{ state }}.d/toolchain_{{ state }}.sh"                        # [unix]
-    - if not exist %PREFIX%\\etc\\conda\\{{ state }}.d\\toolchain_{{ state }}.bat exit 1  # [win]
-    {% endfor %}
+outputs:
+  - name: toolchain
+    # for windows, this is a placebo package.  The default conda-build activation stuff is all we use.
+    # we just don't want people to have to put selectors on their toolchain dep.
+    script: install-toolchain.sh      # [unix]
+    test:
+      commands:
+        # Verify the scripts are in-place.
+        {% for state in ["activate", "deactivate"] %}
+        - test -f "${PREFIX}/etc/conda/{{ state }}.d/toolchain_{{ state }}.sh"                        # [unix]
+        {% endfor %}
+        - echo "I'm on windows, and there's nothing to do here."                                      # [win]
+{% if unix %}
+        - source ${PREFIX}/etc/conda/activate.d/toolchain_activate.sh
+        - if [ -z ${CC+x} ]; then echo "CC is unset after activation" && exit 1; else echo "CC is set to '$CC'"; fi
+        - source ${PREFIX}/etc/conda/deactivate.d/toolchain_deactivate.sh
+        - if [ -z ${CC+x} ]; then echo "CC is unset after deactivation "; else echo "CC is set to '$CC' after deactivation" && exit 1; fi
+
+  - name: toolchain_c_{{ target_platform }}
+    requirements:
+      - {{ pin_subpackage('toolchain', exact=True) }}
+    run_exports:                                                     # [linux]
+      strong:                                                        # [linux]
+        - libgcc-ng >=4.9                                            # [linux]
+
+  - name: toolchain_cxx_{{ target_platform }}
+    requirements:
+      - {{ pin_subpackage('toolchain', exact=True) }}
+    run_exports:
+      strong:
+        - libstdcxx-ng >=4.9                                         # [linux]
+        - libcxx >=4.0                                               # [osx]
+
+  - name: toolchain_fort_{{ target_platform }}
+    requirements:
+      - {{ pin_subpackage('toolchain', exact=True) }}
+      - gcc                                                          # [osx]
```

Comment:
Should we consider this for Linux as well? Personally am of two minds about it. Some low-level packages like `scipy` avoid the `gcc` package to avoid some very old issues we use to have regarding the compiler runtime not always being shipped. While the rest of the stack has no problem using `gcc` for Fortran on either platform.

Reply:
If we don't add this for Linux, we should install `gfortran` in the docker image. I didn't want to install gcc because I remember issues about `gcc` not working sometimes due to bugs in it. Shall I send a PR to install gfortran in docker image? or add `gcc` for Linux here?